docs/spec/channel/version_060.md

```diff
@@ -0,0 +1,137 @@
+
+```
+apiVersion: eventing.knative.dev/v1alpha1
+kind: Broker
+metadata:
+  name: default
+```
+
+##### Overriding the channel of a Broker
+
+With the usage of the `spec.channelTemplate`, a broker instance can change its backing Channel:
+
+```
+apiVersion: eventing.knative.dev/v1alpha1
+kind: Broker
+metadata:
+  name: default
+spec:
+  channelTemplate:
+    provisioner:
+      apiVersion: eventing.knative.dev/v1alpha1
+      kind: ClusterChannelProvisioner
+      name: gcp-pubsub
+```
+
+Details about Broker and Trigger can be found [here](../broker/README.md).
+
+### Data Plane
+
+The data plane describes the input and output flow of a _Channel_:
+
+#### Input
+
+The input container must have an HTTP endpoint running on port 80. However, it might be possible to run on a non-standard port, depending on how well calling pieces handle the Addressable contract. It might be possible to support HTTPS. The HTTP request MUST be a POST to the path '/', where anything else is rejected with a `405 Method Not Supported` and `404 Not Found` respectively.
+
+The HTTP body is passed through and all HTTP headers that match an allowed list (e.g. x-b3-*) are passed through (TODO: document them). It is used exclusively in CloudEvents 0.2 HTTP binary content mode.
+
+Currently there is no request validation is performed.  For example:
+ * All conforming CloudEvents must specify a Type, but a Channel will happily pass along a request that does not have a Type.
+ * Channels will send a body of size zero and no headers, essentially a totally blank message.
+
+HTTP status codes that are used:
+ * `202 Accepted` - The request was received and sent to the middleware successfully.
+ * `404 Not Found`:
+    * The request was sent to path other than '/'.
+    * The request was sent to a Channel that the dispatcher did not know about.
+ * `405 Method Not Supported` - The request was not a POST.
+ * `500 Server Error` - All other errors. Such as:
+    * Could not extract the Channel name from the host.
+    * Could not read the entire HTTP body.
+    * Problem sending to the middleware.
+
+   *NOTE:   * There is no authentication or authorization performed. If a request is received, then it is allowed. 
+
+
+#### Output
+
+An HTTP POST is made to the URI that is injected into the Channel's spec field. It is normally just HTTP, to a hostname, with a path of '/'', because that is what Addressable currently supports. It can be an arbitrary URI. However, it almost certainly needs to be HTTP(S), due to the usage of the `http.Client` package in current implementations.
+
+The request is made by taking the original HTTP body combining it with all the passed through headers. The request is made effectively as a CloudEvents 0.2 HTTP binary content mode request. 
+
+Different Channels handle failures in different ways:
+ * gcp-pubsub - Exponential backoff, up to five minutes. Will continue retrying forever.
+ * kafka - Immediate retry, no backoff. Will continue retrying forever.
```

Comment:
Is this right? I don't see it retrying: https://github.com/knative/eventing/blob/master/contrib/kafka/pkg/dispatcher/dispatcher.go#L292

The error is always ignored.

Reply:
This might be something we want to configure genrically as part of the "endpoint" configuration for a Channel or Source.
A related issue is reliability - for any form of reliable delivery then we will need to be able to configure QoS settings on endpoints.

Reply:
Should then retry and other QoS speced for 0.8+? And tested?